File: bankingOnGolang/internal/model/FundsTransfer.go

package model

import (
    "time"
)

//==============================================================
// FundsTransfer Declaration
//==============================================================
type FundsTransfer struct {
    BaseModel
     TransferReference            string
    Amount            Money
    RequestedDate            time.Time
    ExecutionDate            time.Time
    Purpose            string
    FeeAmount            Money
    SourceAccountId         *uint
    SourceAccount           *Account `gorm:"foreignKey:SourceAccountId"`
    DestinationAccountId         *uint
    DestinationAccount           *Account `gorm:"foreignKey:DestinationAccountId"`
    ExternalBeneficiaryId         *uint
    ExternalBeneficiary           *ExternalAccount `gorm:"foreignKey:ExternalBeneficiaryId"`
    InitiatedById         *uint
    InitiatedBy           *Customer `gorm:"foreignKey:InitiatedById"`
     Transactions           []Transaction `gorm:"foreignKey:TransactionsFromFundsTransferId"`
    Method            PaymentMethod
    Status            PaymentStatus

// parent associations as their child

}

File: bankingOnDjango/models/FundsTransfer.py

from django.db import models
from bankingOnDjango.models.PaymentMethod import PaymentMethod
from bankingOnDjango.models.PaymentStatus import PaymentStatus

#======================================================================
# Class FundsTransfer Declaration
#======================================================================
class FundsTransfer (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	transferReference = models.CharField(max_length=200, null=True)
	amountAmount = models.CharField(max_length=64, null=True)
	amountCurrency = models.CharField(max_length=200, null=True)
	requestedDate = models.DateField(null=True)
	executionDate = models.DateField(null=True)
	purpose = models.CharField(max_length=200, null=True)
	feeAmountAmount = models.CharField(max_length=64, null=True)
	feeAmountCurrency = models.CharField(max_length=200, null=True)
	sourceAccount = models.ForeignKey('Account', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	destinationAccount = models.ForeignKey('Account', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	externalBeneficiary = models.ForeignKey('ExternalAccount', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	initiatedBy = models.ForeignKey('Customer', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	transactions = models.ManyToManyField('Transaction',  blank=True, related_name='+')
	method = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in PaymentMethod])
	status = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in PaymentStatus])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.amount
		str = str + self.currency
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "FundsTransfer";
    
	def objectType(self):
		return "FundsTransfer";
